src/wp-includes/functions.php

```diff
@@ -2079,8 +2080,28 @@ function wp_mkdir_p( $target ) {
 		$dir_perms = 0777;
 	}
 
+	/**
+	 * Fires before the directory creation.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $target Full path to attempt to create.
+	 * @param int    $dir_perms Directory permissions.
+	 */
+	do_action( 'before_create_directory', $target, $dir_perms );
+
 	if ( @mkdir( $target, $dir_perms, true ) ) {
 
+		/**
+		 * Fires after the directory is created and the permissions are set.
```

Comment:
I wonder, should this hook actually be placed after the `if` block below, so that the potential `chmod()` call has already completed? What do you think @benni1516?

Reply:
There doesn't seem to be any permissions being set before this point, so I think this hook indeed needs to go between the `if` block below and `return true;`.